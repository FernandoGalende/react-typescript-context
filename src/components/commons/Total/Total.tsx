import React, {Fragment} from 'react';
import './Total.scss';

type TotalSelectedProps = {
	items: number;
	onClick: Function;
	children: React.ReactNode;
	selected: boolean;
};

export const TotalSelect: React.FC<TotalSelectedProps> = ({items = 0, onClick, children, selected}) => (
	<Fragment>
		<div className='total-wrap'>
			<p data-testid='items-test'>{items} items</p>
			{selected ? (
				<div data-testid='button-test' className='button' onClick={() => onClick()}>
					{children}
				</div>
			) : (
				<div data-testid='button-test-select' className='disable-button'>
					{children}
				</div>
			)}
		</div>
	</Fragment>
);
