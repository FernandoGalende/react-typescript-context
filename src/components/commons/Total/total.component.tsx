import React, {Fragment} from 'react';

import './Total.scss';

type TotalSelectedProps = {
	items: number;
	onClick: any;
	buttonText: string;
};

const TotalSelect: React.FC<TotalSelectedProps> = ({items = 0, onClick, buttonText}) => {
	return (
		<Fragment>
			<div className='total-wrap'>
				<p>{items} items</p>
				<div className='button' onClick={() => onClick()}>
					{buttonText}
				</div>
			</div>
		</Fragment>
	);
};

export default TotalSelect;
