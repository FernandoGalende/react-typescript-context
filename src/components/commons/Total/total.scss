@import '../../../styles/partials';
@import "../../../styles/variables";
@import "../../../styles/guide-utils";

.total-wrap {
	box-sizing: border-box;
	display: flex;
	align-items: center;
	justify-content: space-around;
	width: 100%;

	p {
		white-space: nowrap;
	}
}

.button {
	cursor: pointer !important;
	color: $Secondary;
	white-space: nowrap;
}

.disable-button {
	color: $Grey70;
	cursor: default;
	white-space: nowrap;
}
