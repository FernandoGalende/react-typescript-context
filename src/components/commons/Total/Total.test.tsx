import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {TotalSelect} from './Total';

let getByTestId: Function;
let queryByTestId: Function;
let container: HTMLElement;
const mockFuntion = jest.fn();

describe('<Button select={true}/>', () => {
	beforeEach(() => {
		const queries = render(
			<TotalSelect onClick={mockFuntion} items={4} selected={true}>
				click!
			</TotalSelect>
		);
		container = queries.container;
		getByTestId = queries.getByTestId;
		queryByTestId = queries.queryByTestId;
	});

	it('match snapshot', () => {
		expect(container).toMatchSnapshot();
	});

	it('should be exist', () => {
		expect(getByTestId('button-test')).toBeInTheDocument();
	});

	it('should contain click! text', () => {
		expect(getByTestId('button-test')).toHaveTextContent('click!');
	});

	it('should contain 4 items', () => {
		expect(getByTestId('items-test')).toHaveTextContent('4');
	});

	it('click should be fire function', () => {
		const button = getByTestId('button-test');
		fireEvent.click(button);
		expect(mockFuntion).toBeCalled();
	});

	it('click should be fire function', () => {
		const button = getByTestId('button-test');
		fireEvent.click(button);
		fireEvent.click(button);
		expect(mockFuntion).toHaveBeenCalledTimes(3);
	});

	it('should be not exist', () => {
		expect(queryByTestId('button-test-select')).toBeNull();
	});

	it('should be not exist', () => {
		expect(queryByTestId('button-test')).toBeInTheDocument();
	});
});

describe('<Button select={false}/>', () => {
	beforeEach(() => {
		const queries = render(
			<TotalSelect onClick={mockFuntion} items={4} selected={false}>
				click!
			</TotalSelect>
		);
		getByTestId = queries.getByTestId;
		queryByTestId = queries.queryByTestId;
	});

	it('should be exist', () => {
		expect(getByTestId('button-test-select')).toBeInTheDocument();
	});

	it('should have not clickeable button', () => {
		expect(queryByTestId('button-test')).toBeNull();
	});

	it('should be proper disable', () => {
		const $Grey40 = '#97a1b8';
		expect(getByTestId('button-test-select').style.color).not.toEqual($Grey40);
	});
});
