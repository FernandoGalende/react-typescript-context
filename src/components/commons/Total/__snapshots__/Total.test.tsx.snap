// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button select={true}/> match snapshot 1`] = `
<div>
  <div
    class="total-wrap"
  >
    <p
      data-testid="items-test"
    >
      4
       items
    </p>
    <div
      class="button"
      data-testid="button-test"
    >
      click!
    </div>
  </div>
</div>
`;
